indexing-service/src/main/java/io/druid/indexing/common/task/AppenderatorDriverRealtimeIndexTask.java

```diff
@@ -431,19 +434,45 @@ private void publishSegments(
       String sequenceName
   )
   {
-    ListenableFuture<SegmentsAndMetadata> publishFuture = driver.publish(
+    final ListenableFuture<SegmentsAndMetadata> publishFuture = driver.publish(
         publisher,
         committerSupplier.get(),
         Collections.singletonList(sequenceName)
     );
+    final SettableFuture<SegmentsAndMetadata> handoffDoneFuture = SettableFuture.create();
```

Comment:
Could this boilerplate be extracted in some method?

Reply:
in theory yes, but the different places this kind of style is used have slightly different usage patterns, and I didn't want to spend the mental bandwidth balancing between java futures and guava futures (that are also viable across a wide swath of Guava version) that also fit the different use cases. So I just wrote things specific for each use case.

Reply:
I filed https://github.com/druid-io/druid/issues/5415